llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp

```diff
@@ -1193,6 +1193,17 @@ int AMDGPUCodeGenPrepareImpl::getDivNumBits(BinaryOperator &I, Value *Num,
                                             Value *Den, unsigned AtLeast,
                                             bool IsSigned) const {
   const DataLayout &DL = Mod->getDataLayout();
+  if (!IsSigned) {
+    KnownBits Known = computeKnownBits(Num, DL, 0, AC, &I);
+    // We know all bits are used for division for Operand > smax_bitwidth
```

Comment:
Please try to make the comment more precise. What is smax_bitwidth? What does "Operand > smax_bitwidth" really mean?

Reply:
I had meant smax_bitwidth to be signed max of bit width which corresponds to that of Num/Den. Will reword it as such. The operand range that's handled incorrectly is [SignedMax, UnsignedMax].